.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,49 @@
+name: Unit Tests
+
+on:
+  push:
+    branches:
+      - "main"
+      - "*.latest"
+  pull_request:
+  workflow_dispatch:
+
+permissions: read-all
+
+# will cancel previous workflows triggered by the same event and for the same ref for PRs or same SHA otherwise
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ contains(github.event_name, 'pull_request') && github.event.pull_request.head.ref || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  unit:
+    name: Python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Setup environment
+        uses: ./.github/actions/setup-environment
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run unit tests
+        run: hatch run unit-tests
+        shell: bash
+
+      - name: Publish results
+        uses: ./.github/actions/publish-results
+        if: always()
+        with:
+          source-file: "results.csv"
+          file-name: "unit_results"
+          python-version: ${{ matrix.python-version }}
```

Comment:
Why are you doing this?

Reply:
I'm mimicking the original workflow. I can remove this if it's not needed.